pandas/io/sql.py

```diff
@@ -1422,6 +1425,10 @@ def _get_dtype(self, sqltype):
             return date
         elif isinstance(sqltype, Boolean):
             return bool
+        elif isinstance(sqltype, String):
+            if using_string_dtype():
+                return StringDtype(na_value=np.nan)
```

Comment:
I hard-coded the nan value here for backwards compatability, but this might become a code smell in the future. Have we been trying to avoid this pattern @jorisvandenbossche or is it expected for now?

Reply:
It is required - I somehow did not commit the part that uses it in the caller. Stay tuned...